### ISSUE TITLE
State with multiple views with their own controllers not working

### ISSUE THREAD

**Issue by david-vasquez** [angular-ui/ui-router#2637]
I've defined a state "mystate" with two views:
```javascript
views: {
      'content@': {
         templateUrl: 'scripts/app/view/mystate.html',
         controller: 'ctrl1',
         controllerAs: 'vm'
      },
      'second@mystate': {
          templateUrl: 'scripts/app/view/second.html',
          controller: 'ctrl2'
          controllerAs: 'vmSecond'
      }
},
```
The `content` is a ui-view in my `index.html` while `second`is a ui-view in the mystate.html template.
Controller `ctrl1`is correctly loaded and working, while controller `ctrl2` is not working at all.

In the template `second.html` I can only read model properties defined in the first controller `ctrl1`using `vm.some`, and *cannot* access any model attr defined in its own `ctrl2`

I've put a console.log to check if 'ctrl2' is working, but i don't get either any console output, even though the template has been correctly inserted within the defined ui-view.

This is my `ctrl2` code:
```javascript
angular.module('qibudApp')
    .controller('ctrl2', function ($scope, $log) {
        var vmSecond = this;
        $log.info('Hello!!');
        console.log('Bye');
        vmSecond.someatr = "someValue";
    });
```
Is there a configuration I'm missing to enable multiple views with their own controllers? 
could you please point me out to a running sample having this configuration?

Thanks in advance.

David

**Comment by christopherthielen** [comment]
No, you don't have to do anything special to have named views with controllers

Did you know your 'ctrl2' is accessing an undefined variable `vmSecond`?

Your view setup is a little odd... but it should be able to work.  I you post a plunk I'd be happy to take a look.

**Comment by Septh** [comment]
Don't use `vmSecond` in your `ctrl2` controller, use `this` instead. Use `vmSecond` in your templates, eg. `<p>{{ vmSecond.someatr }}</p>`.

**Comment by david-vasquez** [comment]
Thank you for your replies. I've updated the sample code, in fact the following line was missing `var vmSecond = this;` which misled you from the original problem which is: the controller `ctrl2` is not running at all.

I've even tried the usual ng-inlcude way (nested controllers), but not even that solved the problem. For some reason ctrl2 is not running at all, even though it is correctly loaded in the client-side  (included in index.html).

So I had to rollback to the ugly big single controller code, where ctrl2 code is embedded within the parent controller, and also the html is part of the parent template.

It will be difficult to reproduce this problem in plunkr since a simple code will just work as expected, so I just wanted to understand if there is something especial that I'm missing that makes ctrl2 to simply not run.

Any ideas would be appreciated.
